### PR TITLE
detect: do not run tx detection on non established packets

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6775

Describe changes:
- detect: do not run tx detection on non established packets

#10423 rebased without prerequisite SV PR

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 101 | 111 | 109.9% | 

Pipeline 18637

File: src/flow-worker.c [hunk 1]
```diff
@@ -620,29 +620,32 @@ static TmEcode FlowWorker(ThreadVars *tv, Packet *p, void *data)
                 StreamTcpSessionCleanup(p->flow->protoctx);
             }
         } else if (p->proto == IPPROTO_TCP && p->flow->protoctx) {
-            FramesPrune(p->flow, p);
-            FLOWWORKER_PROFILING_START(p, PROFILE_FLOWWORKER_TCPPRUNE);
-            StreamTcpPruneSession(p->flow, p->flowflags & FLOW_PKT_TOSERVER ?
-                    STREAM_TOSERVER : STREAM_TOCLIENT);
-            FLOWWORKER_PROFILING_END(p, PROFILE_FLOWWORKER_TCPPRUNE);
+            if (p->flags & PKT_STREAM_EST) {
```
Comment (victorjulien) [inline src/flow-worker.c:623]: think this can be merged into the check on line 622 for slightly cleaner code

Comment (catenacyber) [reply src/flow-worker.c:623]: nice

**Comment by catenacyber** [comment]
Was replaced by https://github.com/OISF/suricata/pull/10503

**Pull request closed without merging**